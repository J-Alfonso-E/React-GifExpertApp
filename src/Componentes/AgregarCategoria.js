import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AgregarCategoria = ({EnviarCategoria}) => { //Se recibe la funcion desde GifExpertApp

    const [ValorEntrada, SetValor] = useState('');

    const CambioValor = (e) => {
        SetValor(e.target.value);
        //console.log('Evento Cambio de Valor');
    }

    const EnvioCategoria = (e) => {
        e.preventDefault();
        if(ValorEntrada.trim().length > 2){
            
            EnviarCategoria(cats => [ValorEntrada, ...cats,]); //Cuando se recibe una funcion de Hook el Estado puede ser accedido de esta manera y agregar la nueva categoria
            SetValor('');
        }

        
    }

    return (
        <form onSubmit={EnvioCategoria}>
            <h2>Agregar Categoria</h2>
            <p> {ValorEntrada} </p> {/* Valor de Prueba */}
                <input
                type='text'
                value = {ValorEntrada}
                onChange={CambioValor}
                >
                </input>
        </form>
    )
};

AgregarCategoria.propTypes = {
    EnviarCategoria : PropTypes.func.isRequired
}