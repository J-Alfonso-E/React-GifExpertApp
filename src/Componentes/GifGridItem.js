import React from "react";
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    console.log({title, url});

    return (
        <div className='card animate__fadeInDown'>
            <img className='card animate__bounceIn' src={url} alt={title}></img>
            <p className='card'>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}