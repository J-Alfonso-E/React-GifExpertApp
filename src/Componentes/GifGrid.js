import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { ObtGif } from "../Complementos/GifFetch";
import { GifGridItem } from "./GifGridItem";

import PropTypes from 'prop-types';

export const GifGrid = ( {categoria}) => {

    const {data, loading} = useFetchGifs( categoria);

    return (
        <>
        <h3>{categoria}</h3>

        {loading && 'Cargando' } {/* Operador Ternario donde si es true hace lo primero en caso contrario no hace nada */}
        <div className='card-grid animate__bounceIn'>
            
            
                {data.map( img => (
                    <GifGridItem key={img.id} {...img} />
                ))
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}