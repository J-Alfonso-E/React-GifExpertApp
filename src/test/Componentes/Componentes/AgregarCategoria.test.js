import { AgregarCategoria } from "../../../Componentes/AgregarCategoria";
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

describe('Pruebas en EnviarCategoria', () =>{
    const setCategories = jest.fn(); //Funcion de Prueba provista por jest
    let wrapper = shallow(<AgregarCategoria EnviarCategoria={setCategories}/>);


    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AgregarCategoria EnviarCategoria={setCategories}/>);
    });
    test('debe de mostarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostarse correctamente', () => {
        const input = wrapper.find('input');
        const value = 'Algo';
        //console.log('Input: ' + input);
        input.simulate('change', {target: {value}}); // El evento de cambiar se le debe de enviar el evento que es las primeras llaves, dentro del evento va lo que queremos enviar en este caso es el target y del target un dato
        //console.log(wrapper.find('p').text().trim());
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('debe no debe llamarse el evento de cambio de valor', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}}); // forma corta de envar una funcion de un evento
        expect(setCategories).not.toHaveBeenCalled(); //Verifica que no se haiga llamado sin poner algo en el input
    });

    test('Cambio de Categoria y limpieza de caja de texto', () => {
        /*
        1. Simular el Cambio de Valor
        2. simular el submit
        3. Verificar el Envio de Categoria se haiga llamado
        4. el valor input debe ser ''
                                */

        
        const valor = 'Tales Arise'; 
        wrapper.find('input').simulate('change', {target: {value: valor}});
        

        //expect(wrapper.find('p').text().trim()).toBe(valor);

        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        expect(setCategories).toHaveBeenCalled();

        //forma de ver cuantas veces se llama la funcion
        expect(setCategories).toHaveBeenCalledTimes(1);

        //Para saber si se llamo como funcion

        expect(setCategories).toHaveBeenCalledWith( expect.any(Function));

        const input = wrapper.find('input');
        //console.log(input.value);

        expect(input.prop('value')).toBe('');
    });
});