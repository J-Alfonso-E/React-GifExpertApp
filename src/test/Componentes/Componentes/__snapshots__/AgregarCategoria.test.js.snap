// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en EnviarCategoria debe de mostarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Agregar Categoria
  </h2>
  <p>
     
     
  </p>
   
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
